certbot-apache/certbot_apache/dualparser.py

```diff
@@ -185,37 +186,124 @@ def _create_matching_list(self, primary_list, secondary_list):  # pragma: no cov
                 raise AssertionError("Could not find a matching node.")
         return matched
 
-    def find_blocks(self, name, exclude=True):  # pragma: no cover
+    def find_blocks(self, name, exclude=True):
         """
         Performs a search for BlockNodes using both implementations and does simple
         checks for results. This is built upon the assumption that unimplemented
         find_* methods return a list with a single assertion passing object.
         After the assertion, it creates a list of newly created DualBlockNode
         instances that encapsulate the pairs of returned BlockNode objects.
         """
-        pass
 
-    def find_directives(self, name, exclude=True):  # pragma: no cover
+        primary_blocks = self.primary.find_blocks(name, exclude)
```

Comment:
It feels kind of silly to have this much code copy-pasted; how would you feel about creating a helper function that takes the appropriate `find_*` method and the appropriate `Dual*Node` class?

Reply:
This is a good idea, and now implemented.